packages/core/test/test-util/test-services/v4/test-service/TestEntity.ts

```diff
@@ -340,7 +340,7 @@ export namespace TestEntity {
    * Static representation of the [[complexTypeCollectionProperty]] property for query construction.
    * Use to reference this property in query operations such as 'select' in the fluent request API.
    */
-  export const COMPLEX_TYPE_COLLECTION_PROPERTY: CollectionField<TestEntity> = new CollectionField('ComplexTypeCollectionProperty', TestEntity, new TestComplexTypeField('', TestEntity));
+  export const COMPLEX_TYPE_COLLECTION_PROPERTY: CollectionField<TestEntity, typeof TestComplexType> = new CollectionField('ComplexTypeCollectionProperty', TestEntity, TestComplexType);
```

Comment:
question: what is the return value of `typeof TestComplexType `, object?
Is there a way to avoid using `typeof`?

Reply:
Unfortunately I wouldn't know how. `typeof TestComplexType` is literally the type of `TestComplexType`.

Reply:
If we use `typeof TestComplexType`, which returns `object` anyways, I guess we can just use `object` instead of `typeof ABC` for better readability?

Since we'll check it via `isComplexTypeNameSpace` later, it's still safe.

Reply:
You mean `CollectionField<SomeEntity, Object>`? How is that better readable? Also, safe at runtime is not the same as safe at development time.

Reply:
Discussed, to get rid of the `typeof`, we have to use the same way as the constructable of the normal entity. Since we are happy with the current situation of complex type (metadata), let's use the typeof for now.

Reply:
I am also not super happy with the typeof but I guess this is a TS thing. In Java world you put Collection<String> so the class reference. When you compare also:
```
  EntityT extends EntityBase,
  FieldT extends EdmTypeShared<'any'> | ComplexTypeNamespace
```
Why you can put the `TestEntity` directly but for the type you need to include the typeof. Why is `TestComplexType` not a type? I am also confused.

Reply:
After some discussion, I introduced an equivalent to constructable here.